packages/clarity-js/src/data/metadata.ts

```diff
@@ -194,13 +244,16 @@ export function save(): void {
     setCookie(Constant.SessionKey, [data.sessionId, ts, data.pageNum, upgrade, upload].join(Constant.Pipe), Setting.SessionExpire);
 }
 
-function processCallback(upgrade: BooleanFlag) {
+function processCallback(upgrade: BooleanFlag, consentUpdate: boolean = false): void {
     if (callbacks.length > 0) {
         for (let i = 0; i < callbacks.length; i++) {
             const cb = callbacks[i];
-            if (cb.callback && !cb.called && (!cb.wait || upgrade)) {
-                cb.callback(data, !config.lean);
+            if (cb.callback && (!cb.called || (cb.consentInfo && consentUpdate)) && (!cb.wait || upgrade)) {
+                cb.callback(data, !config.lean, cb.consentInfo ? consentStatus : undefined);
                 cb.called = true;
+                if(consentUpdate){
```

Comment:
can we not do !cb.call || consentUpdate below? - what if we have multiple callback for consent.